packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -772,16 +943,50 @@ function commitUnmount(
             renderPriorityLevel > NormalPriority
               ? NormalPriority
               : renderPriorityLevel;
-          runWithPriority(priorityLevel, () => {
-            let effect = firstEffect;
-            do {
-              const destroy = effect.destroy;
-              if (destroy !== undefined) {
-                safelyCallDestroy(current, destroy);
-              }
-              effect = effect.next;
-            } while (effect !== firstEffect);
-          });
+          if (enableProfilerTimer && current.mode & ProfileMode) {
+            runWithPriority(priorityLevel, () => {
+              let effect = firstEffect;
+              do {
+                const destroy = effect.destroy;
+                if (destroy !== undefined) {
+                  // In order to support the Profiler's onCommit and onPostCommit API,
+                  // we need to attribute time spent in cleanup functions to the right effects group.
+                  // Unfortunately for effects with no changed dependencies, this will be NoHookEffect.
+                  //
+                  // TODO (bvaughn, acdlite) Revisit this when we refactor passive effects and unmount.
```

Comment:
This section is kind of gross. I welcome suggestions for improving it!

In `updateEffectImpl`, we tag hooks with equal dependencies as `NoWork`:
https://github.com/facebook/react/blob/cf0081263ced42ddebe0a0d868701890448bfd4b/packages/react-reconciler/src/ReactFiberHooks.js#L939-L942

I think this makes sense in the common case, but it complicates the `onCommit` / `onPostCommit` implementation because I can't know which group to attribute time to when those effects are later being unmounted:
https://github.com/facebook/react/blob/cf0081263ced42ddebe0a0d868701890448bfd4b/packages/react-reconciler/src/ReactFiberCommitWork.js#L780

The cost of not being able to differentiate between the two is just that we'll end up not attributing some of the time spent in cleanup functions. This is probably not a dealbreaker for the API though.